charts/numaflow/templates/role.yaml

```diff
@@ -0,0 +1,277 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  labels:
+    {{- include "numaflow.labels" . | nindent 4 }}
+    app.kubernetes.io/component: dex-server
+    app.kubernetes.io/name: numaflow-dex-server
+    app.kubernetes.io/part-of: numaflow
+  name: {{ .Values.dexServer.name }}
+  namespace: {{ .Release.Namespace }}
+rules:
+  - apiGroups:
+      - ""
+    resources:
+      - secrets
+      - configmaps
+    verbs:
+      - get
+      - list
+      - watch
+---
+apiVersion: rbac.authorization.k8s.io/v1
```

Comment:
Should be able to support both namespace scoped and cluster scoped installation.

Reply:
yes, we can do that based on condition